docs/3_NEW_TEMPLATE.md

```diff
@@ -1,15 +1,406 @@
+
+> *Have you completed the quickstart guide? Deploy your first infrastructure as code project with Cobalt by following the [quickstart guide](./2_QUICK_START_GUIDE.md).*
 
 ## 3.2 Goals and Objectives
-< Outline the goals and takeaways of this walkthrough/lab >
+
+🔲 Demonstrate how to create *Cobalt Modules* and *CIT*s that work for your custom infrastructure scenario.
+
+🔲 Improve your understanding of how to use existing *Cobalt Module*s and *CIT*s so that they can better work for you.
+
+🔲 Feel confident in moving forward to our next recommended section: *[Testing Cobalt Templates](./4_TEMPLATE_TESTING.md).*
 
 ## 3.3 Prerequisites
-< Enumerate any prerequisite walkthroughs that should be completed prior to this, and any technical prerequisites such as developer environment / tools >
 
-## 3.4 Walkthrough
-< Step-by-step instructions for completing this walkthrough >
+| Prereqs | Description |
+|----------|--------------|
+| [Quickstart Guide](./2_QUICK_START_GUIDE.md) | The quickstart guide provides all of the prerequisites you'll need to create your own *CIT* and run it.|
+| [Terraform Modules](https://www.terraform.io/docs/configuration/modules.html) | An introductory understanding of Terraform modules.|
+
+## 3.4 Walkthrough - Create a New Template
+
+*Cobalt Module*s primarily rely on [Terraform Modules](https://www.terraform.io/docs/configuration/modules.html). The primary purpose of a Terraform Module as a feature is to encapsulate parts of your potential infrastructure configuration into reusable units. CIT's instantiate modules. Here's a great example of Cobalt's existing [Azure service-plan](./../infra/modules/providers/azure/service-plan/README.md) module. It's being reused by several out of the box CIT's.
+
+| Cobalt Infrastructure Template | CIT Instantiating an [Azure service-plan](./../infra/modules/providers/azure/service-plan/README.md) Module |
+|----------|----------|
+|[az-hello-world](./../infra/templates/az-hello-world/README.md)| ![image](https://user-images.githubusercontent.com/10041279/67301762-82123500-f4b5-11e9-9bff-8dc07a4fe001.png) |
+|[az-service-single-region](./../infra/templates/az-hello-world/README.md)| ![image](https://user-images.githubusercontent.com/10041279/67302608-bf2af700-f4b6-11e9-9add-846bd2df42be.png) |
+|[az-isolated-service-single-region](./../infra/templates/az-hello-world/README.md)| ![image](https://user-images.githubusercontent.com/10041279/67302203-2ac09480-f4b6-11e9-839f-19d40abd51ae.png) |
+
+The above table demonstrates in a clear way how *Cobalt Infrastructure Templates* reap the natural benefits of the reusability that's offered by *Cobalt Modules*. This is possible because Terraform modules grant *Cobalt modules* reusability as a feature. Let's experience what it's like to create your own *CIT* from scratch by following the below steps:
```

Comment:
IMO, the above list is really nice and should stay within the document, but does not do what is stated in this paragraph.

The benefit of the module, in my mind, is that it distills the otherwise complicated task of properly configuring a set of related resources in a way that is re-usable for many use cases. An example of this could be "configuring a storage container and granting RBAC access to a service principal," or "configuring App Service with a staging slot and also configuring easy auth"

This context is missing, and probably should live higher in the document when you define what a module is and why you might want one when opening up your "box of cobalt"